### PR TITLE
2.4 cp configurable node creation buffer

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: src/configuration/config.h [hunk 1]
```diff
@@ -9,23 +9,26 @@
...
 
-#define RESULTSET_SIZE_UNLIMITED     UINT64_MAX
-#define QUERY_MEM_CAPACITY_UNLIMITED 0
-#define CONFIG_TIMEOUT_NO_TIMEOUT    0
-#define VKEY_ENTITY_COUNT_UNLIMITED  UINT64_MAX
+#define RESULTSET_SIZE_UNLIMITED           UINT64_MAX
+#define QUERY_MEM_CAPACITY_UNLIMITED       0
+#define CONFIG_TIMEOUT_NO_TIMEOUT          0
+#define VKEY_ENTITY_COUNT_UNLIMITED        UINT64_MAX
+#define DELTA_MAX_PENDING_CHANGES_DEFAULT  10000
```
Comment (jeffreylovitz) [inline src/configuration/config.h:16]: `DELTA_MAX_PENDING_CHANGES` not supported in this version.

File: src/configuration/config.c [hunk 2]
```diff
@@ -555,6 +575,33 @@ bool Config_Option_set(Config_Option_Field field, const char *val) {
...
+			long long msb = (sizeof(long long) * 8) - __builtin_clzll(node_creation_buffer);
+			long long set_msb = 1 << (msb - 1);
+
+			// if the value is not a power of 2
+			// (if any bits other than the MSB are 1),
+			// raise it to the next power of 2
+			if((~set_msb & node_creation_buffer) != 0) {
+				node_creation_buffer = 1 << msb;
+			}
+			break;
```
Comment (jeffreylovitz) [inline src/configuration/config.c:600]: Delete this line

File: tests/flow/test_config.py [hunk 3]
```diff
@@ -131,3 +131,97 @@ def test08_config_restore_timeout_to_default(self):
...
+        config_value = 100
+
+        # Set configuration
+        response = redis_con.execute_command("GRAPH.CONFIG SET %s %d" % (config_name, config_value))
+        self.env.assertEqual(response, "OK")
+
+        # Make sure config been updated.
+        response = redis_con.execute_command("GRAPH.CONFIG GET " + config_name)
+        expected_response = [config_name, config_value]
+        self.env.assertEqual(response, expected_response)
```
Comment (jeffreylovitz) [inline tests/flow/test_config.py:206]: Delete, this config is not run-time on this version.

File: tests/flow/test_config.py [hunk 4]
```diff
@@ -131,3 +131,97 @@ def test08_config_restore_timeout_to_default(self):
...
+                pass
+
+        # No configuration can be set to a string
+        for config in ["MAX_QUEUED_QUERIES", "TIMEOUT", "QUERY_MEM_CAPACITY",
+                       "DELTA_MAX_PENDING_CHANGES", "RESULTSET_SIZE"]:
+            try:
+                redis_con.execute_command("GRAPH.CONFIG SET %s invalid" % config)
+                assert(False)
+            except redis.exceptions.ResponseError as e:
+                assert(("Failed to set config value %s to invalid" % config) in str(e))
```
Comment (jeffreylovitz) [inline tests/flow/test_config.py:191]: I would skip this whole test, the error messages are different and `DELTA_MAX_PENDING_QUERIES` doesn't exist on this version.

**Review by jeffreylovitz — CHANGES_REQUESTED** [review]

**Review by swilly22 — DISMISSED** [review]

**Review by swilly22 — APPROVED** [review]

**Pull request merged**